### ISSUE TITLE
Add JSX helper to Handlebars package

### ISSUE THREAD

**Issue by i-like-robots** [Financial-Times/dotcom-page-kit#729]
Andy G recently created an example app which demonstrates JSX/Handlebars interoperability: https://github.com/Financial-Times/dotcom-page-kit/pull/714

As we now encourage new components to be built in JSX it would be useful for us to provide this advertised interoperability out of the box. For example the conversion team wish to create the new version of `n-live-chat` using JSX instead of Handlebars which would currently mean providing a custom helper with the package or implementing a new helper in each app which integrates it.

A more generic helper could be very straightforward, perhaps only:

```js
const React = require('react')
const ReactDOMServer = require('react-dom/server')

module.exports = function jsx({ hash }) {
  let component = require(hash.require)
 
  if (hash.export) {
    component = component[hash.export]
  } 

  // TODO: add safety check on the current context to avoid passing around sensitive data
  const props = { ...this, ...hash }

  return ReactDOMServer.renderToString(
    React.createElement(component, props)
  )
}
```

Usage:

```hbs
{{{#jsx require="@financial-times/n-thingy"}}}
```

**Comment by apaleslimghost** [comment]
is there any reason we wouldn't encourage people to use `x-handlebars` (and/or split that out of x-dash and release it as a standalone package)?

**Comment by i-like-robots** [comment]
I was thinking about your "when should I use x-engine?" chart when writing this issue and I came to the conclusion that `x-handlebars` is to `x-engine` as `x-engine` is to `x-teaser`. So although `x-handlebars` can be used for this use-case I'm concerned about it's coupling to `x-engine` because I think that could be confusing indirection.

For example, if somebody is working on an app that only needs to integrate the new `n-live-chat` it's not easy to explain why they might need to install and configure `x-engine`.

That said... the example helper above could be named more clearly to better scope it and differentiate it from `x-handlebars`:

```diff
-{{{#jsx require="@financial-times/n-thingy"}}}
+{{{#renderReactComponent require="@financial-times/n-thingy"}}}
```

**Comment by apaleslimghost** [comment]
that's a good point, i'd forgotten about x-engine

**Comment by i-like-robots** [comment]
Fixed by #746
